### PR TITLE
[OCPBUGS-23058] OpenShift installer on AWS requires full access on KMS

### PR DESCRIPTION
**Pull request by Chin2691**
[OCPBUGS-23058] OpenShift installer on AWS requires full access on KMS.

Version(s):

Issue:

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:
Need these permissions to be updated in documentation for KMS

### CODE REVIEW

**Comment by mjpytlak** [comment]
@yunjiang29 PTAL. Thank you.

**Comment by yunjiang29** [comment]
@Chin2691, The current change is applied to `Default IAM role permissions for control plane instance profiles` section, which should NOT be changed, as the issue only happened while BYO-KMS enabled, I'd suggest listing these permissions in a separate section, something like: `Optional permissions when installing a cluster with custom KMS key`. 

This change should apply to version 4.12 and above.

Per the [permission check](https://github.com/openshift/installer/blob/882a2943d87442cb4955a127e4cdf3940216d785/pkg/asset/installconfig/aws/permissions.go#L253-L262), follow permissions are required:
```
kms:Decrypt
kms:Encrypt
kms:GenerateDataKey
kms:GenerateDataKeyWithoutPlainText
kms:DescribeKey
kms:RevokeGrant
kms:CreateGrant
kms:ListGrants
```

File: modules/installation-aws-permissions-iam-roles.adoc [hunk 1]
```diff
@@ -58,3 +58,19 @@ The following lists specify the default permissions for control plane and comput
...
+* `kms:Encrypt`
+* `kms:GenerateDataKey`
+* `kms:GenerateDataKeyWithoutPlainText`
+* `kms:DescribeKey`
+* `kms:RevokeGrant`
+* `kms:CreateGrant`
+* `kms:ListGrants`
+* `kms:ListGrants`
+* `kms:ListGrants`
+* `kms:ListGrants`
```
Comment (yunjiang29) [inline modules/installation-aws-permissions-iam-roles.adoc:75]: @Chin2691 There are duplicate `kms:ListGrants` permissions in the list, please double check the list in [permissions.go](https://github.com/openshift/installer/blob/882a2943d87442cb4955a127e4cdf3940216d785/pkg/asset/installconfig/aws/permissions.go#L253-L262), thanks

**Comment by Chin2691** [comment]
@yunjiang29 apologies for my mistake it is as expected now.

**Comment by yunjiang29** [comment]
@Chin2691 the new section should be part of [Required AWS permissions for the IAM user](https://72362--ocpdocs-pr.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-account#installation-aws-permissions_installing-aws-account), you can put it under `Optional permissions for the cluster owner account when installing a cluster on a shared VPC`

**Comment by Chin2691** [comment]
@yunjiang29 I have updated the changes under section "Optional permissions for the cluster owner account when installing a cluster on a shared VPC"

**Comment by yunjiang29** [comment]
KMS part has been updated by https://github.com/openshift/openshift-docs/pull/70926, this PR can be closed now.

**Pull request closed without merging**